### PR TITLE
Users/jack/promote docker image

### LINKED ISSUE #1766: Move docker publish to zowe-promote-publish pipeline
**Is your feature request related to a problem? Please describe.**

Currently docker push defined in Jenkinsfile will be executed with zowe-install-packaging. This gives the pipeline chance to update official latest/amd64/s390x tags without promote/release process.

**Describe the solution you'd like**

1) move the docker push commands to the release pipeline
2) every release, push `amd64` and `s390x` tags
3) then make a "manifest" of latest as well as `$CURRENT_VERISON`
4) push those manifests too

**Additional context**

Commands to update multi-arch latest tag:

```
# docker manifest create ompzowe/zowe-v1-lts:latest --amend ompzowe/zowe-v1-lts:amd64 --amend ompzowe/zowe-v1-lts:s390x
# docker manifest push ompzowe/zowe-v1-lts:latest
```

### PR DESCRIPTION
**Pull request by jackjia-ibm**
Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [x] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [x] Feature 
- [ ] Other... Please describe:

#### Relevant issues

Fixes #1766 

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jackjia-ibm** [comment]
Test passed with build https://wash.zowe.org:8443/view/Zowe/job/zowe-promote-publish/job/users%252Fjack%252Fpromote-docker-image/8/console.

Artifacts uploaded: 
Screen Shot 2020-11-27 at 9 22 33 PM

Docker image uploaded to test account:
Screen Shot 2020-11-27 at 9 22 49 PM

The publish step is very slow:
- 2 docker load steps, each may take about 18-24 minutes,
- 2 docker push steps, each may tale about 3 minutes,
- overall the publish step 47 minutes.

**Review by stevenhorsman — APPROVED** [review]
Looks good to me

File: Jenkinsfile.release [hunk 1]
```diff
@@ -523,6 +555,51 @@ echo ===============================================
...
+            sh "docker tag ompzowe/zowe-v1-lts:s390x ${USERNAME}/zowe-v1-lts:s390x"
+          }
+          publishCommands += "docker push ${USERNAME}/zowe-v1-lts:s390x\n"
+          manifestAmends += " --amend ${USERNAME}/zowe-v1-lts:s390x"
+        }
+
+        // publish images
+        sh publishCommands +
+          "export DOCKER_CLI_EXPERIMENTAL=enabled\n" +
+          "docker manifest create ${USERNAME}/zowe-v1-lts:latest ${manifestAmends}\n" +
```
Comment (1000TurquoisePogs) [inline Jenkinsfile.release:597]: in addition to `latest`, this is a great opportunity to also create version tags from `${params.ZOWE_RELEASE_VERSION}`
that way, latest and v1.18.0 both point to the particular s390x and amd64 releases.

**Review by 1000TurquoisePogs — APPROVED** [review]
Looks like a great automation of what we've done manually. Only comment is the opportunity to implement versioning.

**Comment by jackjia-ibm** [comment]
@1000TurquoisePogs I tried and found I still need to use `docker manifest` to create version tag, otherwise the tag is only applied to amd64, which is my local cpu architecture.

**Pull request merged**